### PR TITLE
Providing a formula for the installation of the docker-machine completions

### PR DESCRIPTION
**Pull request by Ketouem**
I've created a bash completion script for docker-machine (as boot2docker CLI will be soon deprecated on OS X).

### CODE REVIEW

**Comment by Ketouem** [comment]
The build is failing because the slave checks out the formula on the base repo not the forked one hence not finding the new formula. Is there something to do ?

File: docker-machine-completion.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+class VagrantCompletion < Formula
```
Comment (bfontaine) [inline docker-machine-completion.rb]: Please use `DockerMachineCompletion` here.

Comment (Ketouem) [inline docker-machine-completion.rb]: The class has been renamed, thanks.

**Comment by Ketouem** [comment]
Renamed the formula class to `DockerMachineCompletion`.

File: docker-machine-completion.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class DockerMachineCompletion < Formula
+  homepage "https://github.com/Ketouem/docker-machine-completions"
+  url "https://github.com/Ketouem/docker-machine-completions/archive/0.1.tar.gz"
+  sha256 "0b76cace0f71043c768e6ebc84ad424c62e70f2557c70389d7440ca71b1e3482"
+
+  head "https://github.com/Ketouem/docker-machine-completions.git"
+
+  def install
+    bash_completion.install "contrib/bash/completion.sh" => "docker-machine"
```
Comment (bfontaine) [inline docker-machine-completion.rb]: This should be `"docker-machine_completions.sh" => "docker-machine"`. The first part is the name of the file in the downloaded archive, the second one is the name of the installed one.

Comment (Ketouem) [inline docker-machine-completion.rb]: Made the modification, thanks for the instructions :)

**Comment by Ketouem** [comment]
I've added my fork as a remote inside `/usr/local/Library/Taps/homebrew/homebrew-completions` and checked out the branch and I could successfully install and use the script.
```bash
ketouem@Cyrils-MacBook-Pro : /usr/local/Library/Taps/homebrew/homebrew-completions (docker-machine-completion) $ brew install docker-machine-completion
==> Installing docker-machine-completion from homebrew/homebrew-completions
Warning: Formula file is modified!
Building from source because Library/Taps/homebrew/homebrew-completions/docker-machine-completion.rb has local changes
To install from a bottle instead, run with --force-bottle
==> Downloading https://github.com/Ketouem/docker-machine-completions/archive/0.1.tar.gz
==> Downloading from https://codeload.github.com/Ketouem/docker-machine-completions/tar.gz/0.1
######################################################################## 100.0%
==> Caveats
Bash completion has been installed to:
  /usr/local/etc/bash_completion.d
==> Summary
🍺  /usr/local/Cellar/docker-machine-completion/0.1: 4 files, 28K, built in 2 seconds
```

**Comment by bfontaine** [comment]
@Ketouem the issue comes from the test. This is because the completion script uses identifiers that aren’t valid outside of Bash:

```
==> source /usr/local/Cellar/docker-machine-completion/0.1/etc/bash_completion.d/docker-
/usr/local/Cellar/docker-machine-completion/0.1/etc/bash_completion.d/docker-machine: line 60: `_docker-machine_config': not a valid identifier
```

You can fix this by wrapping the test in `bash -c`:

```rb
  test do
    assert_match "-F _docker-machine",
      shell_output("bash -c 'source #{bash_completion}/docker-machine && complete -p docker-machine'")
  end
```

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Comment by bfontaine** [comment]
Thank you for your contribution to Homebrew @Ketouem; we appreciate it! :tada:

**Pull request closed without merging**